tests/drivers/mipi_dsi/api/src/main.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2022, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/mipi_dsi.h>
+#include <ztest.h>
+
+/**
+ * @addtogroup t_mipi_dsi_driver
+ * @{
+ * @defgroup t_mipi_dsi_api test_mipi_dsi_api
+ * @}
+ */
+
+#if DT_HAS_COMPAT_STATUS_OKAY(nxp_imx_mipi_dsi)
+#define MIPI_NODE DT_INST(0, nxp_imx_mipi_dsi)
+#else
+#define MIPI_NODE ""
+#error "Unsupported SoC"
+#endif
+
+#ifndef CHANNEL
+#define CHANNEL 0
+#endif
+
+static const struct device *mipi_dev;
+
+/**
+ * Test the MIPI generic APIs to test read and write API functionality
+ */
+static void test_generic(void)
+{
+	uint8_t rx_buf[2];
+	uint8_t param[2];
+	ssize_t ret;
+
+	param[0] = MIPI_DCS_SET_DISPLAY_ON;
+	ret = mipi_dsi_generic_write(mipi_dev, CHANNEL, param, 1);
+	zassert(ret >= 0, "Failed to write", NULL);
+
+	param[0] = MIPI_DCS_SET_DISPLAY_BRIGHTNESS;
+	param[1] = 200;
+	ret = mipi_dsi_generic_write(mipi_dev, CHANNEL, param, 2);
+	zassert(ret >= 0, "Failed to write", NULL);
+
+	memset(rx_buf, 0, sizeof(rx_buf));
+
+	param[0] = MIPI_DCS_GET_DISPLAY_BRIGHTNESS;
+	ret = mipi_dsi_generic_read(mipi_dev, CHANNEL, param, 1, rx_buf, 2);
+
+	if (ret != -ENOTSUP) {
+		zassert(ret >= 0, "Failed to do a generic read", NULL);
+	}
+
+}
+
+/**
+ * Test the MIPI DCS APIs to test read and write API functionality
+ */
+static void test_dcs(void)
+{
+	uint8_t rx_buf[2];
+	uint8_t param[2];
+	ssize_t ret;
+
+	ret = mipi_dsi_dcs_write(mipi_dev, CHANNEL, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+	zassert(ret >= 0, "Failed to write", NULL);
+
+	param[0] = 200;
+	ret = mipi_dsi_dcs_write(mipi_dev, CHANNEL, MIPI_DCS_SET_DISPLAY_BRIGHTNESS, param, 1);
+	zassert(ret >= 0, "Failed to write", NULL);
+
+	memset(rx_buf, 0, sizeof(rx_buf));
+
+	param[0] = MIPI_DCS_GET_DISPLAY_BRIGHTNESS;
+	ret = mipi_dsi_dcs_read(mipi_dev, CHANNEL, MIPI_DCS_GET_DISPLAY_BRIGHTNESS, rx_buf, 2);
+
+	if (ret != -ENOTSUP) {
+		zassert(ret >= 0, "Failed to do a dcs read", NULL);
+	}
+}
+
+static void test_preconditions(void)
+{
+	zassert_not_equal(mipi_dev, NULL,
+			  "no mipi device");
+}
+
+void test_main(void)
+{
+	mipi_dev = DEVICE_DT_GET(MIPI_NODE);
```

Comment:
```suggestion
	__ASSERT_NO_MSG(device_is_ready(mipi_dev));
```

Reply:
Done

Reply:
not solved in https://github.com/zephyrproject-rtos/zephyr/commit/59e5d9a22ce2588166cae6a452649c2eeae57285

Reply:
@mmahadevan108  forgot to push?

Reply:
I have re-pushed, please check